lib/BackgroundJob/Tasks/CheckRequirementsTask.php

```diff
@@ -62,17 +62,18 @@ public function do(FaceRecognitionContext $context) {
 		$req = new Requirements($context->appManager, $model);
 
 		if (!$req->pdlibLoaded()) {
-			$this->logInfo('PDLib is not loaded. Cannot continue');
-			// todo: convert to exception
-			return;
+			$error_message = "PDLib is not loaded. Cannot continue";
+			$this->logInfo($error_message);
+			throw new \RuntimeException($error_message);
```

Comment:
I do not think it should be an exception.. 
If you are checking something, should tell you if it is correct or not. The exceptions should be when you want to use something, and it fails unexpectedly.. Not when you are checking it ..

Reply:
So, what would be your proposal?
1. To have this `do` function implement bool return type that tells if we should continue or not? I need to double-check how to handle PHP generator functions and return types (as `do` is yielding and already returns generator object, I think I have to fiddle with something like [this](https://secure.php.net/manual/en/generator.getreturn.php))
2. Have our exception type, let's say `UnsatisfiedRequirementException extends LogicException`, and then we catch only `RuntimeException` in `BackgroundService::execute`

My understanding is that you want to avoid exception as control flow mechanism (routine control). And that is fine and I am OK with it, but I don't know what approach is best. Any other ideas except approach 1?

Reply:
My doubt is rather conceptual. I'm thinking about it .. But I'll probably approve this proposal directly.

Reply:
Check now, I am now returning `mixed` object from generators, both Generator and (at the end) bool. Had to be careful on return types in https://github.com/matiasdelellis/facerecognition/pull/51/files#diff-3eec1bf8e625d99e338e741f9c524b9cR112 but it works. If you sign off, let's commit (or you do it)